diff -urN linux-4.7.1/fs/open.c linux-4.7.1_challenge1.6/fs/open.c
--- linux-4.7.1/fs/open.c	2016-08-16 09:35:15.000000000 +0200
+++ linux-4.7.1_challenge1.6/fs/open.c	2020-10-25 12:50:31.393168999 +0100
@@ -1028,6 +1028,7 @@
 
 SYSCALL_DEFINE3(open, const char __user *, filename, int, flags, umode_t, mode)
 {
+	printk(":) Your file %s is being opened!!!", filename);
 	if (force_o_largefile())
 		flags |= O_LARGEFILE;
 
